src/Components/Subscriptions/Form/index.jsx

```diff
@@ -43,103 +49,35 @@ const Form = () => {
-      const res = await fetch(`${process.env.REACT_APP_API_URL}/api/subscriptions/${id}`);
-      const { data } = await res.json();
-      return data;
-    } catch (error) {
-      console.error(error);
-    }
-  };
-
-  const createSub = async () => {
-    try {
-      const res = await fetch(`${process.env.REACT_APP_API_URL}/api/subscriptions/`, {
-        method: 'POST',
-        body: JSON.stringify(currentSub),
-        headers: {
-          'Content-Type': 'application/json'
-        }
+  useEffect(() => {
+    getMembers(dispatch);
+    dispatch(getClasses());
+    if (id !== '') {
+      const editSub = subs.find((sub) => sub._id === id);
+      console.log(editSub);
+      setCurrentSub({
+        memberId: editSub.memberId?._id,
+        classId: editSub.classId?._id,
+        date: editSub.date.slice(0, 10)
       });
-      const data = await res.json();
-      if (!data.error) {
-        onRedirect.state.message = data.message;
-        history.push(onRedirect);
-      } else {
-        setModalErrorText(data.message);
-        setModalError(true);
-      }
-    } catch (error) {
-      console.error(error);
-    }
-  };
-
-  const editSub = async () => {
-    try {
-      const res = await fetch(`${process.env.REACT_APP_API_URL}/api/subscriptions/${id}`, {
-        method: 'PUT',
-        body: JSON.stringify(currentSub),
-        headers: {
-          'Content-Type': 'application/json'
-        }
+      setValues({
+        member: `${editSub.memberId?.lastName}, ${editSub.memberId?.firstName}`,
+        activity: `${editSub.classId?.activityId?.name}, ${editSub.classId?.day}, ${editSub.classId?.hour} hrs`
       });
-      const data = await res.json();
-      if (!data.error) {
-        onRedirect.state.message = data.message;
-        history.push(onRedirect);
-      } else {
-        setModalErrorText(data.message);
-        setModalError(true);
-      }
-    } catch (error) {
-      console.error(error);
-    }
-  };
-
-  useEffect(() => {
-    getMembers();
-    getClasses();
-    if (id) {
-      const getSub = async () => {
-        const editSub = await getSubscription(id);
-        setCurrentSub({
-          memberId: editSub.memberId._id,
-          classId: editSub.classId._id,
-          date: editSub.date.slice(0, 10)
-        });
-        setValues({
-          member: `${editSub.memberId.lastName}, ${editSub.memberId.firstName}`,
-          activity: `${editSub.classId.activityId?.name}, ${editSub.classId.day}, ${editSub.classId.hour} hrs`
-        });
-      };
-      getSub();
     }
     // eslint-disable-next-line react-hooks/exhaustive-deps
-  }, []);
-
-  const handleClick = () => {
-    id ? editSub(id) : createSub();
+  }, [dispatch]);
+
+  const handleClick = async () => {
+    !id ? postSubscriptions(dispatch, currentSub) : updateSubscription(dispatch, currentSub, id);
+    const updatedState = store.getState();
+    const updatedError = updatedState.subscriptions.error;
+    if (!updatedError) {
```

Comment:
This isn't working on the add function, if you have an error it redirects you to the table, that shouldn't happen, if you have an error you can't leave the form page

Reply:
Done, I removed an 'async await' in the last commit xd